app/src/main/java/com/ncnf/organization/Organization.java

```diff
@@ -0,0 +1,178 @@
+package com.ncnf.organization;
+
+import com.google.firebase.firestore.GeoPoint;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import com.ncnf.database.DatabaseResponse;
+import com.ncnf.database.DatabaseService;
+
+import static com.ncnf.Utils.ADDRESS_KEY;
+import static com.ncnf.Utils.ADMIN_KEY;
+import static com.ncnf.Utils.LOCATION_KEY;
+import static com.ncnf.Utils.NAME_KEY;
+import static com.ncnf.Utils.ORGANIZATIONS_COLLECTION_KEY;
+import static com.ncnf.Utils.ORGANIZED_EVENTS_KEY;
+import static com.ncnf.Utils.PHONE_NB_KEY;
+import static com.ncnf.Utils.UUID_KEY;
+import static com.ncnf.utilities.InputValidator.isArrayEmpty;
+import static com.ncnf.utilities.InputValidator.isStringEmpty;
+
+public class Organization {
+
+    private final UUID uuid;
+    private String name;
+    private GeoPoint location;
+    private String address;
+    private String phoneNumber;
+    private List<String> adminIds;
+    private List<String>  events;
+
+    public Organization(String name, GeoPoint location, String address, String phoneNumber, String originalOwner) {
+
+        this(UUID.randomUUID(), name, location, address, phoneNumber, new ArrayList<String>() {{ add(originalOwner); }}, new ArrayList<>());
+    }
+
+    public Organization(UUID uuid, String name, GeoPoint location, String address, String phoneNumber, List<String> adminIds, List<String> events) {
+        if(isArrayEmpty(adminIds) || !checkAdminIds(adminIds)){
+            throw new IllegalArgumentException("Organization should have at least one admin created when created");
+        }
+
+        this.uuid = uuid;
+        this.name = name;
+        this.location = location;
+        this.address = address;
+        this.phoneNumber = phoneNumber;
+        this.adminIds = adminIds;
+        this.events = events;
+    }
+
+    public boolean checkAdminIds(List<String> l){
+        for(String s : l){
```

Comment:
if you want `return l.stream().noneMatch(InputValidator::isStringEmpty);` is a bit cleaner